spec.html

```diff
@@ -35566,16 +35394,30 @@ <h1>Properties of Map Instances</h1>
       <h1>Map Iterator Objects</h1>
       <p>A Map Iterator is an object, that represents a specific iteration over some specific Map instance object. There is not a named constructor for Map Iterator objects. Instead, map iterator objects are created by calling certain methods of Map instance objects.</p>
 
-      <emu-clause id="sec-createmapiterator" aoid="CreateMapIterator">
+      <emu-clause id="sec-createmapiterator" aoid="CreateMapIterator" oldids="sec-properties-of-map-iterator-instances,table-50,table-internal-slots-of-map-iterator-instances">
         <h1>CreateMapIterator ( _map_, _kind_ )</h1>
         <p>The abstract operation CreateMapIterator takes arguments _map_ and _kind_. This operation is used to create iterator objects for Map methods that return such iterators. It performs the following steps when called:</p>
         <emu-alg>
+          1. Assert: _kind_ is ~key+value~, ~key~, or ~value~.
           1. Perform ? RequireInternalSlot(_map_, [[MapData]]).
-          1. Let _iterator_ be ! OrdinaryObjectCreate(%MapIteratorPrototype%, &laquo; [[IteratedMap]], [[MapNextIndex]], [[MapIterationKind]] &raquo;).
-          1. Set _iterator_.[[IteratedMap]] to _map_.
-          1. Set _iterator_.[[MapNextIndex]] to 0.
-          1. Set _iterator_.[[MapIterationKind]] to _kind_.
-          1. Return _iterator_.
+          1. Let _closure_ be a new Abstract Closure with no parameters that captures _map_ and _kind_ and performs the following steps when called:
+            1. Let _entries_ be the List that is _map_.[[MapData]].
+            1. Let _index_ be 0.
+            1. Let _numEntries_ be the number of elements of _entries_.
+            1. Repeat, while _index_ &lt; _numEntries_,
+              1. Let _e_ be the Record { [[Key]], [[Value]] } that is the value of _entries_[_index_].
+              1. Set _index_ to _index_ + 1.
+              1. If _e_.[[Key]] is not ~empty~, then
+                1. If _kind_ is ~key~, let _result_ be _e_.[[Key]].
+                1. Else if _kind_ is ~value~, let _result_ be _e_.[[Value]].
+                1. Else,
+                  1. Assert: _kind_ is ~key+value~.
+                  1. Let _result_ be ! CreateArrayFromList(&laquo; _e_.[[Key]], _e_.[[Value]] &raquo;).
+                1. Perform ? Yield(_result_).
+                1. NOTE: the number of elements in `_entries_` may have changed while execution of this abstract operation was paused by `Yield`.
```

Comment:
```suggestion
                1. NOTE: the number of elements in _entries_ may have changed while execution of this abstract operation was paused by Yield.
```

(Sorry, that was my fault.)